baybe/campaign.py

```diff
@@ -119,19 +112,21 @@ def from_config(cls, config_json: str) -> Campaign:
         Returns:
             The constructed campaign.
         """
+        from baybe.deprecation import compatibilize_config
+
         config = json.loads(config_json)
-        config["searchspace"] = {
-            "parameters": config.pop("parameters"),
-            "constraints": config.pop("constraints", None),
-        }
-        return _config_converter.structure(config, Campaign)
+
+        # Temporarily enable backward compatibility
+        config = compatibilize_config(config)
+
+        return converter.structure(config, Campaign)
 
     @classmethod
     def to_config(cls) -> str:
```

Comment:
how do you envision `to_config` in the context of the new changes? There ar eprob many problems to solve. To is `to_config` still reasonable if `from_config` somehow gets absorbed by `from_json`? Why do we even have this unimplemented function already included here?

Reply:
Yeah,  agree, `to_config` has been hanging around unimplemented way too long already. Let me remove it in this PR and we can bring it back once actually implemented. Concerning its responsibility: we can definitely discuss it! The way I see it is that it basically gives you a minimal config back, instead of a full serialized object. In a sense, it would close the loop for a roundtrip that starts from the serialized side. For example, I you currently create a campaign using `from_product`, you cannot recreate that config from the object currently, because `to_json` would yield the expanded search space object. Having a dedicated `to_config` method would allow to create these minimal configs from existing python objects. Will create a workitem and resolve.